nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/textract/StartTextractJob.java

```diff
@@ -0,0 +1,442 @@
+        .description("The name of the S3 Bucket that contains the document")
+        .required(true)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .defaultValue("${s3.bucket}")
+        .build();
+    static final PropertyDescriptor DOCUMENT_S3_NAME = new PropertyDescriptor.Builder()
+        .name("Document Object Name")
+        .displayName("Document Object Name")
+        .description("The name of the document in the S3 bucket to perform analysis against")
+        .required(true)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .defaultValue("${filename}")
+        .build();
+    static final PropertyDescriptor DOCUMENT_VERSION = new PropertyDescriptor.Builder()
+        .name("Document Version")
+        .displayName("Document Version")
+        .description("The version of the document in the S3 bucket to perform analysis against")
+        .required(false)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .build();
+    static final PropertyDescriptor CLIENT_REQUEST_TOKEN = new PropertyDescriptor.Builder()
+        .name("Client Request Token")
+        .displayName("Client Request Token")
+        .description("A value that will be sent to AWS in order to uniquely identify the request. This prevents accidental job duplication, which could lead to unexpected expenses. For example, if " +
+            "NiFi is restarted and the same FlowFile is sent a second time, this will result in AWS Textract being smart enough not to process the data again. However, if the Processor settings " +
+            "change, this could result in a processing failure. In such a case, this value may need to be changed.")
+        .required(true)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .defaultValue("${uuid}")
+        .build();
+    static final PropertyDescriptor JOB_TAG = new PropertyDescriptor.Builder()
+        .name("Job Tag")
+        .displayName("Job Tag")
+        .description("The Job Tag to include in the request to AWS. This can be used to identify jobs in the completion statuses that are published to Amazon SNS.")
+        .required(false)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .build();
+    static final PropertyDescriptor KMS_KEY_ID = new PropertyDescriptor.Builder()
+        .name("KMS Key ID")
+        .displayName("KMS Key ID")
+        .description("If specified, the key whose ID is provided will be used to encrypt the results before writing the results to S3")
+        .required(false)
+        .addValidator(NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(FLOWFILE_ATTRIBUTES)
+        .build();
+    static final PropertyDescriptor SNS_TOPIC_ARN = new PropertyDescriptor.Builder()
```

Comment:
It's not really clear to me why SNS is needed. If I understand correctly FetchTextractResults uses the textract api to fetch the results and it doesn't use the SNS topic to get notified.